llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -361,12 +361,14 @@ static bool isCommutative(Instruction *I) {
   return I->isCommutative();
 }
 
-/// \returns inserting index of InsertElement or InsertValue instruction,
-/// using Offset as base offset for index.
-static std::optional<unsigned> getInsertIndex(const Value *InsertInst,
-                                              unsigned Offset = 0) {
+template <typename T>
+static std::optional<unsigned> getInsertExtractIndex(const Value *Inst,
```

Comment:
```suggestion
std::optional<unsigned> getInsertExtractIndex(const Value *Inst,
```

A little strange with that `static`, maybe better declares all these helpers in anonymous namespace?

Reply:
eh, not that weird:
```
$ git grep "template <.*> static" llvm | wc -l
     132
```

Reply:
Yes I think you can commit this directly if you like, I don't have strong opinion about this.